src/Command/PelagosMigrateDatasetFilesCommand.php

```diff
@@ -0,0 +1,209 @@
+            ->addOption('qf', null, InputOption::VALUE_NONE, 'Queue Dataset to Filer')
+            ->addOption('ifc', null, InputOption::VALUE_NONE, 'Do not check if dataset file exist on disk.')
+        ;
+    }
+
+    /**
+     * The symfony command execute function.
+     *
+     * @param InputInterface  $input  The Symfony Console Input.
+     * @param OutputInterface $output The Symfony Console Output.
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $io->title('This is is dataset Migration Tool');
+
+        $queueFiler = $input->getOption('qf') ? true : false;
+        $checkFileExist = $input->getOption('ifc') ? true : false;
+
+        $datasetRepository = $this->entityManager->getRepository(Dataset::class);
+
+        $datasets = $datasetRepository->findAll();
+
+        $io->section('Migrating Datasets Files');
+
+        $dataStore = $io->ask('Path to data store:', '/san/data/store');
+
+        $progressBar = new ProgressBar($output, count($datasets));
+        $progressBar->start();
+
+        foreach ($datasets as $dataset) {
+            $udi = $dataset->getUdi();
+
+            $status = $dataset->getDatasetStatus();
+
+            //$io->note(sprintf('UDI = %s.', $udi));
+
+            if ($status === Dataset::DATASET_STATUS_IN_REVIEW) {
+                $datasetSubmission = $dataset->getLatestDatasetReview();
+                $this->setFile($datasetSubmission, $dataStore, $queueFiler);
+                $datasetSubmission = $dataset->getDatasetSubmission();
+                $this->setFile($datasetSubmission, $dataStore, $queueFiler);
+            } else {
+                $datasetSubmission = $dataset->getDatasetSubmission();
+                $this->setFile($datasetSubmission, $dataStore, $queueFiler);
+            }
+
+            $this->entityManager->persist($dataset);
+            $progressBar->advance();
+        }
+
+        $progressBar->finish();
+        $io->newLine(3);
+
+        $io->section('Flushing');
+        $this->entityManager->flush();
+
+        $io->section('Queuing Filer');
+        $this->submissionsListForFiler = array_unique($this->submissionsListForFiler);
+        $io->note(sprintf('Files For Queue %d.', count(($this->submissionsListForFiler))));
+
+        foreach ($this->submissionsListForFiler as $datasetSubmissionId) {
+            $datasetSubmissionFilerMessage = new DatasetSubmissionFiler($datasetSubmissionId);
+            $this->messageBus->dispatch($datasetSubmissionFilerMessage);
+        }
+
+        $io->success('Done!');
+
+        return 0;
+    }
+
+    /**
+     * Set file data for the dataset submission.
+     *
+     * @param DatasetSubmission  $datasetSubmission The Dataset Submission to process.
+     * @param String             $dataStore         The path to the datatore.
+     * @param Boolean            $queueFiler        Should this also be queued to the filer.
```

Comment:
Return type not specified

Reply:
It doesn't return anything, do we put void?